server/app/modules/DurableJobModule.java

```diff
@@ -81,7 +81,8 @@ public DurableJobRegistry provideDurableJobRegistry(
     durableJobRegistry.register(
         DurableJobName.REPORTING_DASHBOARD_MONTHLY_REFRESH,
         persistedDurableJob ->
-            new ReportingDashboardMonthlyRefreshJob(reportingRepository, persistedDurableJob),
+            new ReportingDashboardMonthlyRefreshJob(
+                reportingRepositoryFactory.create(), persistedDurableJob),
```

Comment:
**suggestion (blocking):** Since creating a new instance populates the map of admin to display names, and this module code only runs when the server starts, initializing the instance here will cause a bug where the map of admin to display names will only be updated when the server starts -- if the admin publishes new programs or changes the display name in a program it won't be included in the map.

To fix this, pass the factory into `ReportingDashboardMonthlyRefreshJob` and call `create` each time the job runs.